src/components/data-exploration/differential-expression-tool/DiffExprCompute.jsx

```diff
@@ -121,6 +121,53 @@ const DiffExprCompute = (props) => {
     );
   };
 
+  const renderError = () => {
```

Comment:
Why are you recomputing `canRunDiffExpr` 3 times, once for each if statement? Can you compute it once at the top of `renderError`, save it to a variable and then use that variable in if statements?
Also the name `canRunDiffExpr` is misleading, because one would expect to get a boolean, whereas it looks from the code like it returns some variable. Can you rename it? It also might need some refactoring (see my other comment)